### PR TITLE
Locales: Add Tarifit locale

### PR DESCRIPTION
**Pull request by ipshel**
Please add Tarifit locale. (rif)

For more info about Tarifit:
https://www.ethnologue.com/language/rif

### CODE REVIEW

File: locales/locales.php [hunk 1]
```diff
@@ -2050,6 +2050,16 @@ public function __construct() {
 		$rhg->slug = 'rhg';
 		$rhg->nplurals = 1;
 		$rhg->plural_expression = '0';
+		
+		$rif = new GP_Locale();
+		$rif->english_name = 'Tarifit';
+		$rif->native_name = 'Tarifiyt';
+		$rif->lang_code_iso_639_3 = 'rif';
+		$rif->country_code = 'ma';
+		$rif->wp_locale = 'rif';
```
Comment (amieiro) [inline locales/locales.php]:
```suggestion
```
This should only be set for locales that are officially supported on translate.wordpress.org.

**Review by amieiro — CHANGES_REQUESTED** [review]

**Comment by amieiro** [comment]
I have made some research and I have some updates:

```
$english_name = 'Tarifit';       // https://en.wikipedia.org/wiki/Tarifit
$native_name = 'Tmaziɣt';        // https://dbpedia.org/page/Tarifit https://en.wikipedia.org/wiki/Tarifit
$text_direction = 'ltr';         // Latin, Arabic script, Tifinagh https://en.wikipedia.org/wiki/Tarifit
$lang_code_iso_639_1 = null;
$lang_code_iso_639_2 = null;
$lang_code_iso_639_3 = 'rif';    // https://iso639-3.sil.org/code/rif
$country_code = 'ma';            // https://www.iso.org/obp/ui/#iso:code:3166:MA
$wp_locale;                      // This should only be set for locales that are offically supported on translate.wordpress.org.
$slug = 'rif';                   // https://iso639-3.sil.org/code/rif
$nplurals = 2;                   // https://translations.launchpad.net/+languages/rif
$plural_expression = 'n != 1';   // https://translations.launchpad.net/+languages/rif
$google_code = null;
$preferred_sans_serif_font_family = null;
$facebook_locale = null;
$alphabet = 'latin';             // Latin, Arabic script, Tifinagh https://en.wikipedia.org/wiki/Tarifit
$word_count_type = 'words';
```

File: locales/locales.php [hunk 2]
```diff
@@ -2050,6 +2050,15 @@ public function __construct() {
 		$rhg->slug = 'rhg';
 		$rhg->nplurals = 1;
 		$rhg->plural_expression = '0';
+		
+		$rif = new GP_Locale();
+		$rif->english_name = 'Tarifit';
+		$rif->native_name = 'Tarifiyt';
```
Comment (amieiro) [inline locales/locales.php:2146]:
```suggestion
		$rif->native_name = 'Tmaziɣt';
```
See https://dbpedia.org/page/Tarifit and https://en.wikipedia.org/wiki/Tarifit

File: locales/locales.php [hunk 3]
```diff
@@ -2050,6 +2050,15 @@ public function __construct() {
...
+		$rif->lang_code_iso_639_3 = 'rif';
+		$rif->country_code = 'ma';
+		$rif->slug = 'rif';
+		$rif->nplurals = 2;
```
Comment (amieiro) [inline locales/locales.php]:
```suggestion
```
It is the default value (`public $nplurals = 2;`), so we can remove it.

File: locales/locales.php [hunk 4]
```diff
@@ -2050,6 +2050,15 @@ public function __construct() {
...
+		$rif->plural_expression = 'n > 1';
```
Comment (amieiro) [inline locales/locales.php]:
```suggestion
```
The plural_expression is `n != 1` (the default one), as you can check [here](https://translations.launchpad.net/+languages/rif), so we can remove it.

**Review by amieiro — DISMISSED** [review]

**Comment by ipshel** [comment]
Tmaziɣt or Tamaziɣt is Berber. And many languages which are derived from Berber call the language Tmaziɣt. Riffian call Tarifit tmaziɣt and others too call their language Tmaziɣt/Tamaziɣt. However we have native names of every language beside Tmaziɣt.
I believe _as Riffian_ we use Tarifiyt as native name. 
`
$english_name = 'Tarifit';
$native_name = 'Tarifiyt';
`
or even we use Riffian as English name:
`
$english_name = 'Riffian';
$native_name = 'Tarifit';
`
https://en.wikipedia.org/wiki/Berber_languages
https://en.wikipedia.org/wiki/Central_Atlas_Tamazight

**Comment by amieiro** [comment]
### English name

I think the name is "Tarifit". We have this name in:
- Wikipedia: https://en.wikipedia.org/wiki/Tarifit
- ISO 639-3 page: https://iso639-3.sil.org/code/rif
- Ethnologue: https://www.ethnologue.com/language/rif

### Native name
I used "Tmaziɣt" because the Wikipedia uses this name between parentheses, and they usually use the parentheses for the native name. I saw this pattern with my two mother languages:
- Galego: https://en.wikipedia.org/wiki/Galician_language
- Español: https://en.wikipedia.org/wiki/Spanish_language

and with other languages:
- https://en.wikipedia.org/wiki/Portuguese_language
- https://en.wikipedia.org/wiki/French_language

**Comment by ipshel** [comment]
I understand that you have checked Wikipedia and you are trying to get the most truthfulness information you can have. as I'm trying too to explain to you why Tmaziɣt is not the most accurate word here.

1. Tmaziɣt and Tamaziɣt are the same term in different grammatical case.
2. Tamaziɣt is used to refer to Berber language. which is the mother language of all sub-Berber languages.
3. Imazighen is the term that is used to refer to the Berber people, who are an indigenous ethnic group that lives in North Africa.
4. Imazighen refer to their language as Tamaziɣt for a number of reasons.
5. Tamaziɣt is the term that has been traditionally used to refer to the Berber language.

The term "Tamaziɣt" refers to the Imaziɣen's language when speaking with non-Berber speakers. Among Berber speakers, "Tamaziɣt" specifically refers to Central Atlas Tamazight (tzm). However, Specific sub-Berber language speakers (Tarifit speakers for example) also use the term "Tamaziɣt" to refer to their own language. It is also believed that the native name for the Tarifit language is "Tarifiyt".

https://web.archive.org/web/20120628211208/http://wold.livingsources.org/language/6

File: locales/locales.php [hunk 5]
```diff
@@ -2119,6 +2119,13 @@ public function __construct() {
 		$rhg->slug = 'rhg';
 		$rhg->nplurals = 1;
 		$rhg->plural_expression = '0';
+		
+		$rif = new GP_Locale();
+		$rif->english_name = 'Tarifit';
+		$rif->native_name = 'Tmaziɣt';
```
Comment (amieiro) [inline locales/locales.php]:
```suggestion
		$rif->native_name = 'Tarifiyt';
```

**Comment by amieiro** [comment]
@ipshel Following your comments, I have updated the native name. I think it is ok to be merged. Is it ok to you?

**Review by amieiro — APPROVED** [review]

**Comment by ipshel** [comment]
@amieiro Everything is ok. 
Thank you. I appreciate that.

**Comment by amieiro** [comment]
> @amieiro Everything is ok. Thank you. I appreciate that.

@ipshel Thank you for your time. Sorry for so much change. I just wanted to be sure that the correct information was added to the project.

**Pull request merged**